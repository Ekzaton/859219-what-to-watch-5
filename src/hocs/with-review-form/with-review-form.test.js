import React from "react";
import renderer from "react-test-renderer";
import PropTypes from "prop-types";

import withReviewForm from "./with-review-form";

describe(`withReviewForm snapshot testing`, () => {
  const MockComponent = (props) => {
    const {children} = props;

    return (
      <div>
        {children}
      </div>
    );
  };

  MockComponent.propTypes = {
    children: PropTypes.oneOfType([
      PropTypes.arrayOf(PropTypes.node),
      PropTypes.node
    ]).isRequired,
  };

  const MockComponentWrapped = withReviewForm(MockComponent);

  it(`renders component correctly`, () => {

    const withReviewFormComponent = renderer.create(
        <MockComponentWrapped>
          <React.Fragment/>
        </MockComponentWrapped>
    );

    expect(withReviewFormComponent.toJSON()).toMatchSnapshot();
  });
});
