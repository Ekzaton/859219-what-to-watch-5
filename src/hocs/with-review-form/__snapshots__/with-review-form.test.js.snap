// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withReviewForm snapshot testing renders component correctly 1`] = `<div />`;
